### PR TITLE
[daikon] Spark: An arquillian spark smoke test to get things started for xpass.

### PR DESCRIPTION
**Pull request by jayunit100**
This is a first spark smoke test using a slight variant of the xpaas spark template (adds top level labels).
It tests the Web UI since that is the fastest validation of basic spark startup.
I'm going to add the spark job tests as well.

### CODE REVIEW

File: spark/src/test/java/org/jboss/test/arquillian/ce/spark/SparkTest.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+@RunWith(Arquillian.class)
+@Template(
+        url="https://gist.githubusercontent.com/jayunit100/3fad23e42a22e473fff7a45947bd3102/raw/ccedbb3020d98267753ba758c9f739d587b3aae5/xpass-spark",
+        labels = "app=spark",
+        parameters = {
+                @TemplateParameter(name = "MASTER_NAME", value="jspark"),
+        }
+)
+@RoleBinding(roleRefName = "view", userName = "system:serviceaccount:${kubernetes.namespace}:jdg-service-account")
```
Comment (jayunit100) [inline spark/src/test/java/org/jboss/test/arquillian/ce/spark/SparkTest.java]: obviously we can live without this.  artifact from some experiments i did when the namespace wasnt working.

Comment (alesj) [inline spark/src/test/java/org/jboss/test/arquillian/ce/spark/SparkTest.java]: W/o @RoleBinding annotation?

File: spark/pom.xml [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+                        <artifactId>maven-surefire-plugin</artifactId>
+                        <version>${version.surefire.plugin}</version>
+                        <configuration>
+                            <runOrder>alphabetical</runOrder>
+                            <redirectTestOutputToFile>true</redirectTestOutputToFile>
+                            <includes>
+                                <include>org/jboss/**/*Test.java</include>
+                            </includes>
+                            <excludes>
+                                <exclude>**/JdgQueryTest.java</exclude>
```
Comment (alesj) [inline spark/pom.xml]: Jdg?

File: spark/pom.xml [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+            </activation>
+
+            <properties>
+                <skipTests>true</skipTests>
+            </properties>
+        </profile>
+
+        <!-- Run tests against wildfly -->
+        <profile>
+            <id>jdg</id>
```
Comment (alesj) [inline spark/pom.xml]: Jdg?

File: spark/pom.xml [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+        <groupId>org.jboss.ce.testsuite</groupId>
+        <artifactId>parent-ce</artifactId>
+        <version>1.0.0-SNAPSHOT</version>
+    </parent>
+
+    <!-- Model Version -->
+    <modelVersion>4.0.0</modelVersion>
+
+    <!-- Artifact Configuration -->
+    <artifactId>jdg-ce-testsuite</artifactId>
```
Comment (alesj) [inline spark/pom.xml]: Jdg?

**Comment by alesj** [comment]
Please fix all "jdg" copy/paste refs.

**Comment by jayunit100** [comment]
How do we feel as a first pass about this?  I can commit the rest of the boilerplate once folks get a first look.

Running spark clients from java itself is a little trickier because the java client needs NSS creds.

**Comment by jayunit100** [comment]
Ack: JDG refs, those all need to be changed.

**Comment by alesj** [comment]
>> How do we feel as a first pass about this? I can commit the rest of the boilerplate once folks get a first look.

I mean, if the test pass, and they are useful (aka properly tests diff functionality), it should be OK.

Although, how is Spark related to what we do at CE team?

**Comment by jayunit100** [comment]
ok guys, this is close but not mergable still a few transitive deps that arquillian isnt happy with.
SPARK_USER worked around the main issue i had w/ NSS, so thats sufficient to keep pressing forward.

**Comment by jayunit100** [comment]
update: i have a few unpushed updates to this right now.  just fyi.  Im trying to come up with a way to test this on a new cluster.

File: spark/src/test/java/org/jboss/test/arquillian/ce/spark/SparkTest.java (hunk 1 above)
Comment (jayunit100) [inline spark/src/test/java/org/jboss/test/arquillian/ce/spark/SparkTest.java]: i think so, ?

**Comment by jayunit100** [comment]
cc @mattf. 
This should get us started.  cleaned out WIP stuff.
Working on another PR for the experimental JavaSparkContext stuff. 
cc @rcernich

File: spark/src/test/java/org/jboss/test/arquillian/ce/spark/SparkTest.java [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+import org.jboss.arquillian.junit.InSequence;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(Arquillian.class)
+@Template(
+
+    url = "https://gist.githubusercontent.com/jayunit100/c2e29f30432d346bb8a9fddb6e8d2012/raw/35952fad20ac34b8ee8a1aac80c81ba6c91b2eb7/xpass-spark.yaml",
```
Comment (jayunit100) [inline spark/src/test/java/org/jboss/test/arquillian/ce/spark/SparkTest.java]: Not sure where the permanant home for this should be.  Its an extension of the original template.

**Comment by jayunit100** [comment]
This is ready as a first pass.
@alesj shall we merge?
to run use `-Pspark`

**Pull request merged**